catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/operation/impl/QueryResponseImpl.java

```diff
@@ -285,7 +290,11 @@ public Result take() {
     private Result handleTake() {
         Result result = null;
         try {
-            result = queue.take();
+            if (timeoutMillis <= 0) {
+                result = queue.take();
```

Comment:
That's what I originally had implemented but @rzwiefel pointed out that it does indeed break the contract of the Query interface. I think you were tagged in an earlier comment about it. We should discuss it